crates/signal/src/lib.rs

```diff
@@ -213,37 +359,47 @@ mod tests {
 
     use super::*;
 
+    #[cfg(unix)]
     pub fn raise_signal(kind: SignalKind) {
         // Safety: This is a test, and we control the process.
         unsafe {
             libc::raise(kind.as_raw_value());
         }
     }
 
+    #[cfg(windows)]
+    pub fn raise_signal(kind: SignalKind) {
+        unsafe {
+            use winapi::um::winconGenerateConsoleCtrlEvent;
+
+            GenerateConsoleCtrlEvent(kind.as_raw_value(), 0);
+        }
+    }
+
     #[cfg(not(valgrind))] // test is time-sensitive
     #[tokio::test]
     async fn signal_handler() {
-        let mut handler = SignalHandler::with_signals([SignalKind::user_defined1()])
-            .with_signal(SignalKind::user_defined2())
-            .with_signal(SignalKind::user_defined1());
+        let mut handler = SignalHandler::with_signals([UnixSignalKind::user_defined1()])
+            .with_signal(UnixSignalKind::user_defined2())
+            .with_signal(UnixSignalKind::user_defined1());
 
-        raise_signal(SignalKind::user_defined1());
+        raise_signal(SignalKind::Unix(UnixSignalKind::user_defined1()));
 
         let recv = (&mut handler).with_timeout(Duration::from_millis(5)).await.unwrap();
 
-        assert_eq!(recv, SignalKind::user_defined1(), "expected SIGUSR1");
+        assert_eq!(recv, SignalKind::Unix(UnixSignalKind::user_defined1()), "expected SIGUSR1");
 
         // We already received the signal, so polling again should return Poll::Pending
         let recv = (&mut handler).with_timeout(Duration::from_millis(5)).await;
 
         assert!(recv.is_err(), "expected timeout");
 
-        raise_signal(SignalKind::user_defined2());
+        raise_signal(SignalKind::Unix(UnixSignalKind::user_defined2()));
 
         // We should be able to receive the signal again
         let recv = (&mut handler).with_timeout(Duration::from_millis(5)).await.unwrap();
 
-        assert_eq!(recv, SignalKind::user_defined2(), "expected SIGUSR2");
+        assert_eq!(recv, UnixSignalKind::user_defined2(), "expected SIGUSR2");
     }
```

Comment:
these tests will not work on windows, we should disable them on windows and add a new windows only test.